kaniko_test.go

```diff
@@ -280,3 +284,230 @@ func TestTarPathValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestLocalTarballPush(t *testing.T) {
+	tests := []struct {
+		name          string
+		localTarPath  string
+		repo          string
+		autoTag       bool
+		tags          []string
+		commitRef     string
+		repoBranch    string
+		expectedError bool
+		expectedTags  []string
+		mockLoadErr   error
+		mockPushErr   error
+	}{
+		{
+			name:          "empty_repo_fails",
+			localTarPath:  "/path/to/image.tar",
+			repo:          "",
+			expectedError: true,
+		},
+		{
+			name:          "nonexistent_tarball_fails",
+			localTarPath:  "/path/that/does/not/exist/image.tar",
+			repo:          "test-repo",
+			expectedError: true,
+		},
+		{
+			name:          "load_image_fails",
+			localTarPath:  createTestTarball(t),
+			repo:          "test-repo",
+			expectedError: true,
+			mockLoadErr:   fmt.Errorf("load failed"),
+		},
+		{
+			name:          "push_image_fails",
+			localTarPath:  createTestTarball(t),
+			repo:          "test-repo",
+			expectedError: true,
+			expectedTags:  []string{"latest"},
+			mockPushErr:   fmt.Errorf("push failed"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockPlugin := Plugin{
+				Build: Build{
+					SourceTarPath:   tt.localTarPath,
+					Repo:            tt.repo,
+					Tags:            tt.tags,
+					AutoTag:         tt.autoTag,
+					DroneCommitRef:  tt.commitRef,
+					DroneRepoBranch: tt.repoBranch,
+				},
+				LoadImageFromTarball: func(path string) (v1.Image, error) {
+					if tt.mockLoadErr != nil {
+						return nil, tt.mockLoadErr
+					}
+					return &mockImage{}, nil
+				},
+				PushImageToRegistry: func(img v1.Image, dest string) error {
+					if tt.mockPushErr != nil {
+						return tt.mockPushErr
+					}
+					return nil
+				},
+			}
+
+			err := mockPlugin.Exec()
+
+			if tt.expectedError {
+				if err == nil {
+					t.Errorf("Expected an error, but got none")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Unexpected error: %v", err)
+				}
+			}
+		})
+	}
+}
+
+// Mock image type to simulate v1.Image
+type mockImage struct{}
+
+func (m *mockImage) Size() (int64, error) {
+	return 0, nil
+}
+
+func (m *mockImage) RawConfigFile() ([]byte, error) {
+	return nil, nil
+}
+
+func (m *mockImage) Digest() (v1.Hash, error) {
+	return v1.Hash{}, nil
+}
+
+func (m *mockImage) Manifest() (*v1.Manifest, error) {
+	return nil, nil
+}
+
+func (m *mockImage) RawManifest() ([]byte, error) {
+	return nil, nil
+}
+
+func (m *mockImage) LayerByDigest(hash v1.Hash) (v1.Layer, error) {
+	return nil, nil
+}
+
+func (m *mockImage) LayerByDiffID(hash v1.Hash) (v1.Layer, error) {
+	return nil, nil
+}
+
+func (m *mockImage) Layers() ([]v1.Layer, error) {
+	return nil, nil
+}
+
+func (m *mockImage) MediaType() (types.MediaType, error) {
+	return "", nil
+}
+
+func (m *mockImage) ConfigFile() (*v1.ConfigFile, error) {
+	return nil, nil
+}
+
+func (m *mockImage) ConfigName() (v1.Hash, error) {
+	return v1.Hash{}, nil
+}
```

Comment:
let's move this to a dedicated test util file `mock_crane.go` and define mock functions `MockCraneLoad` `MockCranePush` for tests - then this mocks can be directly used in kaniko_test.go

Reply:
Done. ✅